ports/qttools/portfile.cmake

```
@@ -92,15 +92,18 @@ qt_install_submodule(PATCHES    ${${PORT}_PATCHES}
                     )
 
 if(VCPKG_TARGET_IS_OSX)
-    set(OSX_APP_FOLDERS Designer.app Linguist.app pixeltool.app)
+    set(OSX_APP_FOLDERS Designer.app Linguist.app pixeltool.app qdbusviewer.app)
     if (FEATURE_qdbus)
         message(STATUS "Built qdbusviewer")
         list(APPEND OSX_APP_FOLDERS qdbusviewer.app)
     endif()
     foreach(_appfolder IN LISTS OSX_APP_FOLDERS)
-        message(STATUS "Moving: ${_appfolder}")
-        file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/tools/${PORT}/bin/${_appfolder}")
-        file(RENAME "${CURRENT_PACKAGES_DIR}/bin/${_appfolder}/" "${CURRENT_PACKAGES_DIR}/tools/${PORT}/bin/${_appfolder}/")
+        # Folders are only existing in case of native builds 
+        if(EXISTS "${CURRENT_PACKAGES_DIR}/bin/${_appfolder}")
```

This condition may hide problems from misconfigured prerequisites (`FEATURE_qdbus`?). It should be avoided as much as possible.

---

Do you have a suggestion? Checking VCPKG_CROSSCOMPILING does not work, because it will also indicate cross compiling if the tools are generated.

---

@daschuer 
For this modification:
Benefit:
The modified code adds an existence check for each folder in the `OSX_APP_FOLDERS` list before attempting to move them. 

This change can prevent unexpected errors or exceptions that may occur when trying to move or create a directory that does not exist. Therefore, this modification can improve the robustness and reliability of the code.

Bad:
Condition should not be used: `FEATURE_ qdbus`

A more reliable condition needs to be used. 
Similar to this effect
```
if(DEFINED FEATURE_qdbus AND FEATURE_qdbus)
```

---

> This change can prevent unexpected errors or exceptions that may occur when trying to move or create a directory that does not exist. Therefore, this modification can improve the robustness and reliability of the code.

No longer noticing when an expected directory exists isn't "more robust".

The key is to know when it should exist. I don't really understand why it shouldn't exist for cross-builds.

---

They do not exist, because the Qt build system does not build them. They are useless anyway, because the target architecture builds can't be used on the host. A host build of Qt is a prerequisite.

I am sure the build process will fail early If something wrong. But if you wish to have here a test for the Qt build system, please contribute it. 
I have no idea how to do it reliable.